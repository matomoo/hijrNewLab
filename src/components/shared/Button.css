@import '../../root.css';

.btn {
  display: flex;
  width: 85vw;
  height: 100%;
  padding: 20px;
  border-radius: 3px;
  border-style: solid;
  border-width: 2px;
  margin-bottom: 8px;

  justify-content: center;
  align-items: center;
  cursor: pointer;
  opacity: 1;
  transition: opacity 0.2s ease-in-out;

  &:hover {
    opacity: 0.5;
  }
  &:active {
    opacity: 0.5;
  }

  @media (--desktop) {
    width: 320px;
    height: 50px;
  }
}

.txt {
  font-size: 1em;
}

.logoImg {
  position: absolute;
  left: 16px;
  height: 20px;
  width: 20px;
  object-fit: cover;
}