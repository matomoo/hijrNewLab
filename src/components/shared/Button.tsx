import React, { Component } from 'react';

const classes = require('./Button.css');

export class Button extends Component<any, any> {
  public render() {
    return (
      <div
        className={classes.btn}
        style={
          this.props.white
            ? { borderColor: '#fff', color: '#fff' }
            : { borderColor: '#000', color: '#000' }
        }
        onClick={() => this.props.onClick()}
      >
        {
          this.props.imgSrc
            ? <img
                src={this.props.imgSrc}
                srcSet={this.props.srcset}
                className={classes.logoImg}
              />
            : null
        }
        <span
          className={classes.txt}
          >{this.props.txt}</span>
      </div>
    );
  }
}

export default Button;
