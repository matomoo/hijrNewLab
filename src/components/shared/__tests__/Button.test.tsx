import React from 'react';
import { shallow } from 'enzyme';

import { Button } from '../Button';

// const component = shallow(
//     <Button white={true} btnTxt='Button 1st test' />
//   );

// test for the pure component
describe('Button shared component test', () => {
  const component = shallow(
    <Button white={true} btnTxt='Button 2nd test' />,
  );

  it('component and snapshot matches', () => {
    expect(component).toMatchSnapshot();
  });

  it('component onClick function has to be added and work to make the count 2', () => {
    let count = 1;
    const onClick = () => {
      count++;
    };

    component.setProps({ onClick: () => onClick() });
    expect(component).toMatchSnapshot();

    component.find('div').simulate('click');
    expect(count).toBe(2);
  });
});
