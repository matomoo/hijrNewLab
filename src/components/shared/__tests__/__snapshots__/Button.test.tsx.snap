// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button shared component test component and snapshot matches 1`] = `
<div
  className="btn"
  onClick={[Function]}
  style={
    Object {
      "borderColor": "#fff",
      "color": "#fff",
    }
  }
>
  <span
    className="txt"
  />
</div>
`;

exports[`Button shared component test component onClick function has to be added and work to make the count 2 1`] = `
<div
  className="btn"
  onClick={[Function]}
  style={
    Object {
      "borderColor": "#fff",
      "color": "#fff",
    }
  }
>
  <span
    className="txt"
  />
</div>
`;
